### PR TITLE
[public-api] Implement experimental TeamsService.CreateTeam

### PR DESCRIPTION
**Pull request by easyCZ**
## Description

Implements TeamsService.CreateTeam by delegating to server.

## Related Issue(s)

* Depends on https://github.com/gitpod-io/gitpod/pull/14151
* Depends on https://github.com/gitpod-io/gitpod/pull/14170
* Depends on https://github.com/gitpod-io/gitpod/pull/14273
* Part of https://github.com/gitpod-io/gitpod/issues/14309

UI uses this RPC in 
* https://github.com/gitpod-io/gitpod/pull/14232

## How to test

1. Open up preview, log-in
2. Obtain a token by running the below in the browser console
	```javascript
	await window._gp.gitpodService.server.generateNewGitpodToken({ type: 1, scopes: ["function:getGitpodTokenScopes",
		"function:createTeam",
		"function:getTeamMembers",
		"function:getGenericInvite",
		"resource:default",]})
	```
3. Set your credentials on the command line:
	```bash
	BEARER_TOKEN="<token>"
	```
5. Hit the endpoint, adjusting the `<team_name>` variable
```bash
	curl \
    --header 'Content-Type: application/json' \
    --header "Authorization: Bearer $BEARER_TOKEN" \
    --data '{ "name": "<team_name>" }' \
    https://api.mp-papi-crf9cf921e3c.preview.gitpod-dev.com/gitpod.experimental.v1.TeamsService/CreateTeam

{
  "team": {
    "id": "7f8c0ccc-a7a3-42bd-b4ce-81b8f7b5e51b",
    "name": "foobar2",
    "slug": "foobar2",
    "members": [
      {
        "userId": "a7561431-c58d-4c73-830f-0efe83a42387",
        "role": "TEAM_ROLE_OWNER"
      }
    ],
    "teamInvitation": {
      "id": "redacted"
    }
  }
}
```

## Release Notes

```release-note
NONE
```

## Werft options:

- [ ] /werft with-local-preview
      If enabled this will build `install/preview`
- [x] /werft with-preview
- [ ] /werft with-large-vm
- [ ] /werft with-integration-tests=all
      Valid options are `all`, `workspace`, `webapp`, `ide`

### CODE REVIEW

**Review by andrew-farries — APPROVED** [review]
/hold for nit

tested and works 👍

**Comment by laushinka** [comment]
Was testing without reloading, turns out @andrew-farries was faster 🚀
But yes, also works!

File: components/public-api-server/pkg/apiv1/team.go [hunk 1]
```diff
@@ -0,0 +1,97 @@
...
+	connectionPool proxy.ServerConnectionPool
+
+	v1connect.UnimplementedTeamsServiceHandler
+}
+
+func (s *TeamService) CreateTeam(ctx context.Context, req *connect.Request[v1.CreateTeamRequest]) (*connect.Response[v1.CreateTeamResponse], error) {
+	token := auth.TokenFromContext(ctx)
+
+	if req.Msg.GetName() == "" {
+		return nil, connect.NewError(connect.CodeInvalidArgument, fmt.Errorf("Name is a required argument when creating a team."))
```
[the remark this answers is not in the document]
Comment (easyCZ) [reply components/public-api-server/pkg/apiv1/team.go:38]: I made this one uppercase as it's a public facing error. I guess we don't really have any best practices around this yet but I find the user-facing error message starting with capitals a bit friendlier.

**Pull request merged**